File: Programing_Basics_with_JS/whileLoopExercise/05.Coins.js
function coins(input) {
    let moneyInBgn = Number(input[0]);

    let money = Math.floor(moneyInBgn * 100); 
    let coins = 0;

    while(money > 0) {
        coins++;
        if (money >= 200) {
            money -= 200;
        } else if (money >= 100) {
            money -= 100;
        } else if (money >= 50) {
            money -= 50;
        } else if (money >= 20) {
            money -= 20;
        } else if (money >= 10) {
            money -= 10;
        } else if (money >= 5) {
            money -= 5;
        } else if (money >= 2) {
            money -= 2;
        } else {
            money -= 1;
        }
    }

    console.log(coins);
}

coins(['1.23'])
coins(['2'])
coins(['0.56'])
coins(['2.73'])

File: Programming_Basics_with_Python/while_loop_exercise/05._coins.py
def bg_coins_count(money):
    bgn, coins = str(money).split('.')
    all_coins = int(bgn) * 100 + int(coins)
    coins_count = 0

    while not all_coins == 0:
        if all_coins >= 200:
            new_coins = all_coins // 200
            coins_count += new_coins
            all_coins -= new_coins * 200
        elif all_coins >= 100:
            new_coins = all_coins // 100
            coins_count += new_coins
            all_coins -= new_coins * 100
        elif all_coins >= 50:
            new_coins = all_coins // 50
            coins_count += new_coins
            all_coins -= new_coins * 50
        elif all_coins >= 20:
            new_coins = all_coins // 20
            coins_count += new_coins
            all_coins -= new_coins * 20
        elif all_coins >= 10:
            new_coins = all_coins // 10
            coins_count += new_coins
            all_coins -= new_coins * 10
        elif all_coins >= 5:
            new_coins = all_coins // 5
            coins_count += new_coins
            all_coins -= new_coins * 5
        elif all_coins >= 2:
            new_coins = all_coins // 2
            coins_count += new_coins
            all_coins -= new_coins * 2
        elif all_coins >= 1:
            new_coins = all_coins // 1
            coins_count += new_coins
            all_coins -= new_coins * 1

    return coins_count


money = float(input())

print(bg_coins_count(money))